### ISSUE TITLE
Conditional boolean attributes

### ISSUE THREAD

**Issue by bennypowers** [11ty/webc#155]
Consider this lit-html template
```js
html`<a href="${ifDefined(href)}></a>`;
```
if `href` is undefined, the output will be:
```html
<a></a>
```

Why is this important?

Consider

```html
<script webc:for="script of scripts"
      webc:keep
      :defer="script.defer"
      :async="script.async"
      :src="url(script.src)"></script>
```
In this case, all scripts will be `async` and `defer` because those are boolean attributes whose presence indicates true.

If we wanted to use a template and ifs:

```html
<template webc:nokeep webc:for="script of scripts">
  <script webc:if="defer && async" defer async :src="url(script.src)"></script>
  <script webc:elseif="defer" defer            :src="url(script.src)"></script>
  <script webc:elseif="async" async            :src="url(script.src)"></script>
  <script webc:else                            :src="url(script.src)"></script>
</template>
```

I guess that's ok? If you have lots of attrs, you have to keep track of a whole tree of if/then/else.

But what if `null` meant "remove this attribute"?:

```html
<script webc:for="script of scripts"
       :defer="script.defer ?? null"
       :async="script.async ?? null"
       :src="url(script.src)"></script>
```

**Comment by darthmall** [comment]
I agree that it would be nice to have a way to conditionally set attributes on an element like this. Other uses for this might be conditionally setting `hidden` or `inert` on elements.

As of WebC 0.11, I would tend to solve this problem with a render function:

```html
<script webc:type="js">
  let attrs = [];
  if (script.defer) attrs.push("defer");
  if (script.async) attrs.push("async");

  `<script ${attrs.join(" ")} src="${url(script.src)}"></script>`
</script>
```

But I bet this would be a common enough task that making it easy to do from markup would be ideal.

**Comment by zachleat** [comment]
`@attributes` is another option here, fwiw!

https://www.11ty.dev/docs/languages/webc/#@attributes

maybe something like:

```html
<img @attributes="({ ...(script.defer ? { defer: "" } : {}) })">
```

**Comment by darthmall** [comment]
That’s pretty clever, but it’s rather arcane, I think. With the spread operator and all of those parentheses, it’s not at all obvious to me what’s happening there, and I know what I’m looking at. I’d still favor a more direct way to handle boolean attributes on elements.

**Comment by bennypowers** [comment]
`false` is the key?

**Comment by darthmall** [comment]
I think I discovered another inconsistency with `false`, `true`, and `''` for attributes: when assigned to to `class`, `false` becomes a string (`class="false"`) and `''` causes the attribute to be removed entirely.

**Comment by zachleat** [comment]
I think `class` and `style` are special cases here, because we merge the values between host and component definition. There is some special code in there to prevent superfluous `class=""` and `style=""` attributes from appearing in the output unnecessarily.

But worth filing if you think it merits another look!
